### ISSUE TITLE
Dependency on codemirror-graphql branch

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-dashboard#1633]
Make sure these boxes are checked before submitting your issue -- thanks for reporting issues back to Parse Dashboard!

- [x] You're running version >=1.0.23 of Parse Dashboard.
- [x] You're running version >=2.3.2 of Parse Server.
- [x] You've searched through [existing issues](https://github.com/ParsePlatform/parse-dashboard/issues?utf8=%E2%9C%93&q=). Chances are that your issue has been reported or resolved before.

#### Environment Setup
Parse Dashboard has a dependency on a personal branch:
https://github.com/parse-community/parse-dashboard/blob/faf8d2dc739625f183c8b548b8a3d09b079276a7/package.json#L41

The branch is not maintained.

#### Solution
The dependency should point to the [original repository](https://github.com/graphql/graphiql/tree/main/packages/codemirror-graphql) or Parse should to maintain its own fork if necessary.

**Comment by Nes-si** [comment]
Seems like this issue makes it impossible using **parse-dashboard** in project if you have new npm version.
`npm i` finishes with an error:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: graphql@15.4.0
npm ERR! node_modules/graphql
npm ERR!   graphql@"15.4.0" from parse-server@4.5.0
npm ERR!   node_modules/parse-server
npm ERR!     parse-server@"^4.5.0" from the root project
npm ERR!   peer graphql@"^14.0.0 || ^15.0.0" from graphql-language-service-parser@1.8.0
npm ERR!   node_modules/graphql-language-service-parser
npm ERR!     graphql-language-service-parser@"^1.0.18" from codemirror-graphql@0.6.12
npm ERR!     node_modules/codemirror-graphql
npm ERR!       codemirror-graphql@"github:timsuchanek/codemirror-graphql#details-fix" from parse-dashboard@2.1.0
npm ERR!       node_modules/parse-dashboard
npm ERR!         parse-dashboard@"^2.1.0" from the root project
npm ERR!     graphql-language-service-parser@"^1.2.2" from graphql-language-service-interface@1.3.2
npm ERR!     node_modules/codemirror-graphql/node_modules/graphql-language-service-interface
npm ERR!       graphql-language-service-interface@"^1.0.18" from codemirror-graphql@0.6.12
npm ERR!       node_modules/codemirror-graphql
npm ERR!         codemirror-graphql@"github:timsuchanek/codemirror-graphql#details-fix" from parse-dashboard@2.1.0
npm ERR!         node_modules/parse-dashboard
npm ERR!   4 more (graphql-language-service-types, apollo-link, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer graphql@"^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0" from graphql-language-service-interface@1.3.2
npm ERR! node_modules/codemirror-graphql/node_modules/graphql-language-service-interface
npm ERR!   graphql-language-service-interface@"^1.0.18" from codemirror-graphql@0.6.12
npm ERR!   node_modules/codemirror-graphql
npm ERR!     codemirror-graphql@"github:timsuchanek/codemirror-graphql#details-fix" from parse-dashboard@2.1.0
npm ERR!     node_modules/parse-dashboard
npm ERR!       parse-dashboard@"^2.1.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by beppemac** [comment]
I solved by deleting the `node_modules` folder and `package-lock.json` file and finally run `npm i`.

**Comment by Nes-si** [comment]
> I solved by deleting the `node_modules` folder and `package-lock.json` file and finally run `npm i`.

It's not a solution. The problem deals with new npm version (v7) which installs peer-dependencies. You obviously used npm v6 or earlier.

**Comment by moneppo** [comment]
Originally I got stuck believing that `--legacy-peer-deps` would work until this package was upgraded. I was able to downgrade to `npm@16.14.11` to work around this issue.

**Comment by ritterb82** [comment]
Same problem.  Can't run parse dashboard with latest npm version.

**Comment by ritterb82** [comment]
> Originally I got stuck believing that `--legacy-peer-deps` would work until this package was upgraded. I was able to downgrade to `npm@16.14.11` to work around this issue.

Just wanted to point out the version should be 6.14.11.  ( not 16.14.11 )

**Comment by architectcodes** [comment]
Just weighing in to let you know this makes for a pretty disappointing first impression when getting started with Parse Server and the stable version of Node.js and npm 😬

**Comment by mtrezza** [comment]
Thanks everyone for pinning this down. I think we should add a [compatibility table](https://github.com/parse-community/parse-server#compatibility) just like we have for Parse Server and ideally add CI tests to detect such incompatibilities.

**Comment by mtrezza** [comment]
For convenience, this is how to downgrade from npm 7 to 6 on macOS using brew:
```
$ brew unlink node
$ brew link --overwrite --force node@14
```

**Comment by stage88** [comment]
Would it work if we upgrade dependancy `graphql-playground-react`. It seems to be the one which breaks the install. See my comment here https://github.com/parse-community/parse-server/issues/7090#issuecomment-789659616

I dont have much experience with graphql, but I can try tomorrow. Its getting late here in AUS.

**Comment by mtrezza** [comment]
Thanks for trying this out. If we can't find a solution, we may have to either fork and maintain our own to bypass the prepare script or remove the playground all together.

**Comment by elijahjcobb** [comment]
Can confirm this is still an issue with the latest versions of everything.

**Comment by rakuemoevi** [comment]
I can confirm i'm not able to deploy to heroku getting error message:
/usr/bin/git ls-remote -h -t ssh://git@github.com/timsuchanek/codemirror-graphql.git
remote:        npm ERR! 
remote:        npm ERR! Host key verification failed.
remote:        npm ERR! fatal: Could not read from remote repository.

**Comment by architectcodes** [comment]
Node 14 and lower is not compatible with Apple Silicon chips (see https://github.com/nodejs/node/issues/36161). It’s a deadlock – Parse Dashboard cannot be used on late 2020 Macs and newer 😱

**Comment by mtrezza** [comment]
I would think the dashboard runs on Node 15 too, have you tried? Is this related to this issue or something separate?

**Comment by architectcodes** [comment]
Node 15 ships with npm 7, which refuses to install parse-dashboard.

If you’re suggesting we can downgrade npm to v6 on Node 15, I didn’t try that, because it’s a very painful workflow that I wouldn’t want to introduce. Everyone on the team would need to remember to switch to a non-default version of npm after installing Node.

**Comment by danielsanfr** [comment]
Don't close it please.

**Comment by mtrezza** [comment]
> Node 14 and lower is not compatible with Apple Silicon chips (see nodejs/node#36161). It’s a deadlock – Parse Dashboard cannot be used on late 2020 Macs and newer 😱

Isn't that also true for Parse Server? I think it currently also doesn't install with npm 7 due to dependency issues.

**Comment by alljinx** [comment]
> > Node 14 and lower is not compatible with Apple Silicon chips (see [nodejs/node#36161](https://github.com/nodejs/node/issues/36161)). It’s a deadlock – Parse Dashboard cannot be used on late 2020 Macs and newer 😱
> 
> Isn't that also true for Parse Server? I think it currently also doesn't install with npm 7 due to dependency issues.
> 
> Can anyone try and confirm this?

No problem installing Parse Server with npm 7 but no way for Parse Dashboard even with the lagacy peer deps or the force option. With regret, we are forced to consider discontinuing the use of this package. It's sad, especially just for one dependency...

**Comment by dblythy** [comment]
Closed via #1807
